ECS: support composite region_iso_code (#206)
logstash-plugins/logstash-filter-geoip: docs/index.asciidoc

                                                    support, but populated
                                                    with 2-character country
                                                    code_

| `postal_code`      | `[geo][postal_code]`       | `98106`
| `region_name`      | `[geo][region_name]`       | `Washington`
| `region_code`      | `[geo][region_code]`       | `WA`
| `region_iso_code`* | `[geo][region_iso_code]`   | `US-WA`
| `timezone`         | `[geo][timezone]`          | `America/Los_Angeles`
| `location`*        | `[geo][location]`          | `{"lat": 47.6062, "lon": -122.3321}"`
| `latitude`         | `[geo][location][lat]`     | `47.6062`
| `longitude`        | `[geo][location][lon]`     | `-122.3321`

| `domain`           | `[domain]`                 | `example.com`

| `asn`              | `[as][number]`             | `98765`
| `as_org`           | `[as][organization][name]` | `Elastic, NV`

| `isp`              | `[mmdb][isp]`              | `InterLink Supra LLC`
| `dma_code`         | `[mmdb][dma_code]`         | `819`
| `organization`     | `[mmdb][organization]`     | `Elastic, NV`

> |===========================

Reviewer: Love the table. Nice presentation and it renders well, too.

NOTE: `*` indicates a composite field, which is only populated if GeoIP lookup result contains all components.

==== Details

When using a City database, the enrichment is aborted if no latitude/longitude pair is available.

> The `location` field combines the latitude and longitude into a structure called http://geojson.org/geojson-spec.html[GeoJSON].

Reviewer: This link (http://geojson.org/geojson-spec.html) resolves, but it says the document is obsolete. It recommends https://datatracker.ietf.org/doc/html/rfc7946 instead. Should we update or point to something else?

> The `location` field combines the latitude and longitude into a structure called http://geojson.org/geojson-spec.html[GeoJSON].

Author: ```suggestion
The `location` field combines the latitude and longitude into a structure called https://datatracker.ietf.org/doc/html/rfc7946[GeoJSON].
```

Author: IETF standards track is better than a private `.org` in my book 👍🏼

> When using a default <<plugins-{type}s-{plugin}-target>>, the templates provided by the {logstash-ref}/plugins-outputs-elasticsearch.html[elasticsearch output] maps the field to an {ref}/geo-point.html[Elasticsearch Geo_point datatype].

Reviewer: ```suggestion
When you are using a default <<plugins-{type}s-{plugin}-target>>, the templates provided by the {logstash-ref}/plugins-outputs-elasticsearch.html[elasticsearch output] map the field to an {ref}/geo-point.html[Elasticsearch Geo_point datatype].
```

Reviewer: In this construction, "templates" becomes the subject, which doesn't really make sense. Suggesting "you..." or perhaps "the output..." 
The verb "map" should agree with "templates" (instead of elasticsearch output).

===== `database`

  * Value type is <<path,path>>
  * If not specified, the database defaults to the `GeoLite2 City` database that ships with Logstash.

The path to MaxMind's database file that Logstash should use.
The default database is `GeoLite2-City`.
This plugin supports several free databases (`GeoLite2-City`, `GeoLite2-Country`, `GeoLite2-ASN`)
and a selection of commercially-licensed databases (`GeoIP2-City`, `GeoIP2-ISP`, `GeoIP2-Country`).

> Database auto-update applies to default distribution.

Reviewer: ```suggestion
Database auto-update applies to the default distribution.
```

> When `database` points to user's database path, auto-update will be disabled.

Reviewer: ```suggestion
When `database` points to user's database path, auto-update is disabled.
```

An array of geoip fields to be included in the event.

Possible fields depend on the database type.
By default, all geoip fields from the relevant database are included in the event.

For a complete list of available fields and how they map to an event's structure, see <<plugins-{type}s-{plugin}-field-mapping,field mapping>>.

[id="plugins-{type}s-{plugin}-ecs_compatibility"]
===== `ecs_compatibility`

* Value type is <<string,string>>
* Supported values are:
** `disabled`: unstructured geo data added at root level

> ** `v1`, `v8`: uses fields that are compatible with Elastic Common Schema (for example, `[client][geo][country_name]`; see <<plugins-{type}s-{plugin}-field-mapping,field mapping>>)

Reviewer: ```suggestion
** `v1`, `v8`: use fields that are compatible with Elastic Common Schema. Example: `[client][geo][country_name]`. See <<plugins-{type}s-{plugin}-field-mapping,field mapping>> for more info. 
```

Reviewer: Minor tweak for easier reading and translation
